### PR TITLE
prow: add pilot postsubmit to configuration.

### PR DESCRIPTION
**Pull request by nlandolfi**
This introduces a `pilot-postsubmit` job.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by nlandolfi** [comment]
Corresponding pilot PR: https://github.com/istio/pilot/pull/940

**Comment by nlandolfi** [comment]
I removed the max_concurrency for now, as discussed.

File: prow/config.yaml [hunk 1]
```diff
@@ -153,3 +153,51 @@ presubmits:
...
+  istio/pilot:
+  - name: pilot-postsubmit
+    branches:
+    - master
+    spec:
+      containers:
+      - image: gcr.io/istio-testing/prowbazel:0.1.5
+        args:
+        - "--branch=$(PULL_REFS)"
+        - "--upload=gs://kubernetes-jenkins/logs"
```
Comment (sebastienvas) [inline prow/config.yaml]: Are you sure this is fine ? I don t think you need this.

File: prow/config.yaml [hunk 2]
```diff
@@ -153,3 +153,51 @@ presubmits:
...
+        - "--git-cache=/root/.cache/git"
```
Comment (sebastienvas) [inline prow/config.yaml]: What does git cache do ?

**Review by sebastienvas — CHANGES_REQUESTED** [review]

File: prow/config.yaml (hunk 2 above)
Comment (nclando) [reply prow/config.yaml]: it caches the git repo so you may not need to pull all the objects again, but I mistakenly included the flag.

File: prow/config.yaml (hunk 1 above)
Comment (nclando) [reply prow/config.yaml]: Definitely not -- my mistake. removed.

**Review by sebastienvas — APPROVED** [review]

**Review by yutongz — APPROVED** [review]

**Comment by yutongz** [comment]
When you want get this pr merged, comment "/test all" to re-trigger test.

**Pull request merged**